lib/autokey/interface.py

```diff
@@ -671,116 +678,121 @@ def __sendString(self, string):
-                            self.__releaseKey(Key.SHIFT)
-                            char_as_hex_string = '{:X}'.format(ord(char))
-                            for hex_char in char_as_hex_string:
-                                hexKeyCodeList = self.localDisplay.keysym_to_keycodes(ord(hex_char))
-                                hexKeyCode, hexOffset = self.__findUsableKeycode(hexKeyCodeList)
-                                self.__sendKeyCode(hexKeyCode)
-                            self.__pressKey(Key.ENTER)
-                        else:
-                            self.__sendKeyCode(keyCode, theWindow=focus)
-                    if offset == 1:
-                        self.__pressKey(Key.SHIFT)
-                        self.__sendKeyCode(keyCode, self.modMasks[Key.SHIFT], focus)
-                        self.__releaseKey(Key.SHIFT)
-                    if offset == 4:
-                        self.__pressKey(Key.ALT_GR)
-                        self.__sendKeyCode(keyCode, self.modMasks[Key.ALT_GR], focus)
-                        self.__releaseKey(Key.ALT_GR)
-                    if offset == 5:
-                        self.__pressKey(Key.ALT_GR)
-                        self.__pressKey(Key.SHIFT)
-                        self.__sendKeyCode(keyCode, self.modMasks[Key.ALT_GR]|self.modMasks[Key.SHIFT], focus)
-                        self.__releaseKey(Key.SHIFT)
-                        self.__releaseKey(Key.ALT_GR)
-
-                elif char in self.remappedChars:
-                    keyCode, offset = self.remappedChars[char]
-                    if offset == 0:
-                        self.__sendKeyCode(keyCode, theWindow=focus)
-                    if offset == 1:
-                        self.__pressKey(Key.SHIFT)
-                        self.__sendKeyCode(keyCode, self.modMasks[Key.SHIFT], focus)
-                        self.__releaseKey(Key.SHIFT)
-                else:
-                    logger.warning("Unable to send character %r", char)
+                self.__send_keycode_for_char(char, focus)
             except Exception as e:
                 logger.exception("Error sending char %r: %s", char, str(e))
 
         self.__ignoreRemap = False
 
+    def __send_keycode_for_char(self, char, focus):
+        # Offset encodes the modifiers needed to convert the base key press
+        # into the desired result symbol.
+        keyCode, offset = self.__get_usable_char_keycode_and_offset(char)
+        if keyCode is not None:
+            if offset == 0:
+                if self.localDisplay.lookup_string(ord(char)) is None:
+                    # No reasonable translation of key to string found
+                    self.__sendByTypingAsUnicodePoint(char, focus)
+                else:
+                    self.__sendKeyCode(keyCode, theWindow=focus)
+            if offset == 1:
+                self.__send_keycode_with_modifiers_pressed(
+                    keyCode, Key.SHIFT, focus)
```

Comment:
looks like this introduced the issue, the __send_keycode_with_modifiers_press function is expecting a list of modifiers that it can iterate through, passing a single `Key.SHIFT` value will have the for loop treat it as a string